packages/python/plotly/plotly/tests/test_optional/test_px/test_trendline.py

```diff
@@ -182,28 +189,41 @@ def test_ols_trendline_slopes():
         ("ewm", dict(alpha=0.5)),
     ],
 )
-def test_trendline_on_timeseries(mode, options):
-    df = px.data.stocks()
+def test_trendline_on_timeseries(constructor, mode, options):
+    df = nw.from_native(constructor(px.data.stocks().to_dict(orient="list")))
```

Comment:
@emilykl for whatever reason, here changing:

```diff
- def test_trendline_on_timeseries(constructor, mode, options):
-     df = nw.from_native(constructor(px.data.stocks().to_dict(orient="list")))
+ def test_trendline_on_timeseries(backend, mode, options):
+     df = nw.from_native(px.data.stocks(return_type=backend))
```

ends up not resulting in an error. I will investigate further why that's the case.

All other tests that use plotly data are switched to use `return_type` directly

Reply:
Update: `stocks` dataset `"date"` column was automatically detected as date type for pyarrow. I added an explicit conversion in the loader to make sure that if not asked to be datetime, then it will be of string type. Then this test is passing with the diff in the above comment